translate/filters/checks.py

```diff
@@ -55,14 +55,16 @@
         %(                          # initial %
         (?P<boost_ord>\d+)%         # boost::format style variable order, like %1%
         |
+        \|?                         # boost::format allows to enclose params in pipes, %|spec|
               (?:(?P<ord>\d+)\$|    # variable order, like %1$s
               \((?P<key>\w+)\))?    # Python style variables, like %(var)s
         (?P<fullvar>
-            [+#-]*                  # flags
+            [+#\- ]*                # flags
```

Comment:
@unho this may look like unrelated fix but they are all related, this is first thing I did and then I had to fix others otherwise it won't work properly, because printf test will fail this string: `Down payment is %d%% something`. It will match '% s' as placeholder and fail, while it is actually part of `%%` and it is a valid string. I wanted to fix everything related to that percent things at once, because this sole fix will introduce a problem immediately (we have a string that is like that for example).